### PR TITLE
FEAT: Adding support for stream returning completion results

### PR DESCRIPTION
**Pull request by takatost**
Currently, although i can pass the stream parameter and stream the response, it's actually still blocking and waits for the complete result before streaming the output. To achieve non-blocking streaming, i need to combine it with the stream parameter of requests.

### CODE REVIEW

**Comment by jiayini1119** [comment]
Hi, thanks for the contribution!

File: xinference/client.py [hunk 1]
```diff
@@ -282,13 +282,15 @@ def generate(
             for key, value in generate_config.items():
                 request_body[key] = value
 
-        response = requests.post(url, json=request_body)
+        stream = bool(generate_config and generate_config.get("stream"))
```
Comment (UranusSeven) [inline xinference/client.py:285]: Get may result in a KeyError if stream is absent.

File: xinference/client.py [hunk 2]
```diff
@@ -390,13 +392,15 @@ def chat(
...
```
Comment (UranusSeven) [inline xinference/client.py:395]: Ditto.

File: xinference/client.py [hunk 3]
```diff
@@ -452,13 +456,15 @@ def chat(
...
```
Comment (UranusSeven) [inline xinference/client.py:459]: Ditto.

**Review by UranusSeven — APPROVED** [review]

**Pull request merged**